ui/components/app/auto-detect-nft/auto-detect-nft-modal.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback, useContext } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  Box,
+  Text,
+  ModalBody,
+  ModalFooter,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AlignItems,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextAlign,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { setUseNftDetection } from '../../../store/actions';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { getProviderConfig } from '../../../ducks/metamask/metamask';
+import { getCurrentLocale } from '../../../ducks/locale/locale';
+import { ORIGIN_METAMASK } from '../../../../shared/constants/app';
+
+type AutoDetectNftModalProps = {
+  isOpen: boolean;
+  onClose: (arg: boolean) => void;
+};
+function AutoDetectNftModal({ isOpen, onClose }: AutoDetectNftModalProps) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const { chainId } = useSelector(getProviderConfig);
+
+  const handleNftAutoDetection = useCallback(
+    (val) => {
+      trackEvent({
+        event: val
+          ? MetaMetricsEventName.NftAutoDetectionEnableModal
+          : MetaMetricsEventName.NftAutoDetectionDisableModal,
+        category: MetaMetricsEventCategory.Navigation,
+        properties: {
+          chain_id: chainId,
+          locale: getCurrentLocale,
```

Comment:
same fix you found for the token modal.  although i think locale not necessary.  seems to get automatically added to events [here](https://github.com/MetaMask/metamask-extension/blob/92e6b945eb54fafba491d9a84ca95c2e5fbaf932/app/scripts/controllers/metametrics.js#L761)?

Reply:
Definitely, will remove, also, i think i this modal might be removed, based on some latest design discussions